ddtrace/internal/writer.py

```diff
@@ -417,13 +429,13 @@ def flush_queue(self, raise_exc=False):
 
             encoded = self._encoder.join_encoded(enc_traces)
             try:
-                self._send_payload(encoded, len(enc_traces))
-            except (compat.httplib.HTTPException, OSError, IOError):
+                self._retry_upload(self._send_payload, encoded, len(enc_traces))
+            except tenacity.RetryError as e:
                 self._metrics_dist("http.errors", tags=["type:err"])
                 self._metrics_dist("http.dropped.bytes", len(encoded))
                 self._metrics_dist("http.dropped.traces", len(enc_traces))
                 if raise_exc:
-                    raise
+                    e.reraise()
```

Comment:
There's also a `reraise=True` in tenacity if that fits you better.

Reply:
Ah thanks, I missed that from the docs! I think I'll keep it as it is so to avoid having the same set of exception classes in two different places.